src/marqo/s2_inference/model_downloading/from_hf.py

```diff
@@ -0,0 +1,48 @@
+from marqo.tensor_search.models.external_apis.hf import HfAuth, HfModelLocation
+from typing import Optional
+from huggingface_hub import hf_hub_download
+from marqo.s2_inference.logger import get_logger
+from huggingface_hub.utils._errors import RepositoryNotFoundError
+from marqo.s2_inference.errors import ModelDownloadError
+
+logger = get_logger(__name__)
+
+
+def download_model_from_hf(
+        location: HfModelLocation,
+        auth: Optional[HfAuth] = None,
+        download_dir: Optional[str] = None):
+    """Downloads a pretrained model from HF, if it doesn't exist locally. The basename of the
+    location's filename is used as the local filename.
+
+    hf_hub_download downloads the model if it does not yet exist in the cache.
+
+    Args:
+        location: repo_id and filename to be downloaded.
+        auth: contains HF API token for model access
+        download_dir: [not yet implemented]. The location where the model
+            should be stored
+
+    Returns:
+        Path to the downloaded model
+    """
+    if download_dir is not None:
+        logger.warning(
+            "Hugging Face model download was given the `download_dir` argument, "
+            "even though it is not yet implemented. "
+            "The specified model will be downloaded but the `download_dir` "
+            "parameter will be ignored."
+        )
+    download_kwargs = location.dict()
+    if auth is not None:
+        download_kwargs = {**download_kwargs, **auth.dict()}
+    try:
+        return hf_hub_download(**download_kwargs)
```

Comment:
We currently don't do this for other HF models: https://github.com/marqo-ai/marqo/blob/mainline/src/marqo/s2_inference/hf_utils.py#LL23C40-L23C55

Unless I am missing something. It would be good to add this, at some point. Do you think it's best to do this in this request, or as part of a new/different PR?

Reply:
https://github.com/marqo-ai/marqo/pull/466
i have this PR. yes, i think we can do this later.